code/game/objects/structures/altar.dm

```diff
@@ -0,0 +1,80 @@
+/*
+ *Altars and Podiums
+ *Altars should later have other functions but I am not coder enough to do them yet.
+ *
+ */
+
+/obj/structure/altar
+	name = "altar"
+	desc = "Some sort of spooky altar used by the chaplain for, something. Best not to fiddle with it."
+	icon = 'icons/obj/DeusSpritus.dmi'
+	icon_state = "altar2_on"
+	density = 1
+	anchored = 1
+	var/on = 1
+	light_range = 4
+	light_color = LIGHT_COLOR_ORANGE
+	var/last_check = 0
+
+/obj/structure/altar/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/structure/altar/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+
+/obj/structure/altar/attack_hand(mob/M as mob)
+
+	if(on == 0)
+		on = 1
+		update_icon()
+		set_light(4)
+
+	else
+		on = 0
+		update_icon()
+		set_light(0)
+
+/obj/structure/altar/update_icon()
+	if (on)
+		icon_state = "altar2_on"
```

Comment:
This is a great time to learn ternary statements. Get rid of the if entirely and use

`icon_state = "alter2_[on ? "on" : "off"]"`

Reply:
`on ? value-if-true : value-if-false` btw

Reply:
ok that makes sense